### PR TITLE
[environment_joint_state] added joint state updater for environment

### PR DESCRIPTION
**Pull request by sunava**
For real robot also the environment as state updater so we update the believe state with giskard

### CODE REVIEW

File: src/pycram/ros_utils/joint_state_updater.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        :param joint_state_topic: Name of the joint state topic, needs to publish sensor_msgs/JointState
+        """
+        self.tf_listener = tf.TransformListener()
+        time.sleep(1)
+        self.tf_topic = tf_topic
+        self.joint_state_topic = joint_state_topic
+        self.tf_timer = create_timer(Duration().from_sec(0.1), self._subscribe_tf)
+        self.joint_state_timer = create_timer(Duration().from_sec(0.1), self._subscribe_joint_state)
+
+        atexit.register(self._stop_subscription)
```
Comment (tomsch420) [inline src/pycram/ros_utils/joint_state_updater.py:39]: that is some very nice thing here, we need that at more classes :+1:

File: src/pycram/ros_utils/joint_state_updater.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        """
+        Stops the Timer for TF and joint states and therefore the updating of the robot in the world.
+        """
+        self.tf_timer.shutdown()
+        self.joint_state_timer.shutdown()
+
+
+class EnvironmentStateUpdater:
+    """
+    Updates the environment in the Bullet World with information of the real environment published to ROS topics.
```
Comment (tomsch420) [inline src/pycram/ros_utils/joint_state_updater.py:75]: is this really bullet world dependent or just abstract world dependent?

File: src/pycram/ros_utils/joint_state_updater.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        in the rospy implementation.
+
+        :param msg: JointState message published to the topic.
+        """
+        try:
+            msg = rospy.wait_for_message(self.joint_state_topic, JointState)
+            for name, position in zip(msg.name, msg.position):
+                try:
+                    # Attempt to get the joint state. This might throw a KeyError if the joint name doesn't exist
+                    if World.environment.get_joint_state(name) is None:
```
Comment (tomsch420) [inline src/pycram/ros_utils/joint_state_updater.py:110]: this does not look bullet world dependent. Please update the docstring

**Review by tomsch420 — CHANGES_REQUESTED** [review]
Please provide tests

File: src/pycram/ros_utils/joint_state_updater.py
[the remark this answers is not in the document]
Comment (sunava) [reply src/pycram/ros_utils/joint_state_updater.py:1]: maybe then object_joint_state?

**Comment by sunava** [comment]
No test for this as discussed on 28.11.2024 with @tomsch420

**Pull request merged**